File: frontend/src/app/core/models/video-metadata-dto.ts
export interface VideoMetadataDto {
    id: string
    title: string
    description: string
    thumbnail: string
    authorId: string
    authorNickname: string
    viewCount: number
    tags: string[]
    visibility: string
    processingProgress: string
    uploadDate: Date
    editDate: Date
    duration: string
}
  

File: backend/YouTubeV2.Application/DTO/VideoMetadataDTOS/VideoMetadataDto.cs
using YouTubeV2.Api.Enums;
using YouTubeV2.Application.Enums;

namespace YouTubeV2.Application.DTO.VideoMetadataDTOS
{
    public record VideoMetadataDto(
        Guid id,
        string title,
        string description,
        Uri thumbnail,
        string authorId,
        string authorNickname,
        int viewCount,
        IReadOnlyCollection<string> tags,
        Visibility visibility,
        ProcessingProgress processingProgress,
        DateTimeOffset uploadDate,
        DateTimeOffset editDate,
        string duration);
}
